ragflow_deps/download_go_deps.py

```
@@ -143,19 +159,68 @@ def download_with_progress(url, filename):
+                continue
+
         if os.path.isdir(target):
             print(f"  ✓ {subdir} already extracted to {target}")
             continue
         os.makedirs(target, exist_ok=True)
         print(f"  Extracting {archive} → {target}")
-        with tarfile.open(archive_path) as tf:
-            tf.extractall(target)
+        if archive_path.endswith(".zip"):
+            with zipfile.ZipFile(archive_path) as zf:
+                zf.extractall(target)
```

Ruff (0.16.3)

[error] Uses of `tarfile.extractall()`

(S202)